src/main/java/org/folio/circulation/resources/RequestCollectionResource.java

```diff
@@ -179,6 +190,53 @@ void empty(RoutingContext routingContext) {
       .thenAccept(r -> r.writeTo(routingContext.response()));
   }
 
+  void move(RoutingContext routingContext) {
+    WebContext context = new WebContext(routingContext);
+    Clients clients = Clients.create(context, client);
+
+    JsonObject representation = routingContext.getBodyAsJson();
+
+    String id = getRequestId(routingContext);
+
+    final RequestRepository requestRepository = RequestRepository.using(clients);
+    final RequestQueueRepository requestQueueRepository = RequestQueueRepository.using(clients);
+
+    final ItemRepository itemRepository = new ItemRepository(clients, true, true, true);
+    final LoanRepository loanRepository = new LoanRepository(clients);
+    final LoanPolicyRepository loanPolicyRepository = new LoanPolicyRepository(clients);
+
+    final MoveRequestService moveRequestService = new MoveRequestService(
+      RequestRepository.using(clients),
+      requestQueueRepository,
+      new RequestPolicyRepository(clients),
+      UpdateRequestQueue.using(clients),
+      new UpdateItem(clients),
+      new UpdateLoanActionHistory(clients),
+      new UpdateLoan(clients, loanRepository, loanPolicyRepository),
+      loanRepository,
+      itemRepository);
+
+    requestRepository.getById(id)
+      .thenApply(r -> r.map(RequestAndRelatedRecords::new))
+      .thenApply(r -> r.map(rr -> asMove(rr, representation)))
+      .thenComposeAsync(r -> r.after(moveRequestService::moveRequest))
+      .thenApply(r -> r.map(RequestAndRelatedRecords::getRequest))
+      .thenApply(r -> r.map(new RequestRepresentation()::extendedRepresentation))
+      .thenApply(OkJsonResponseResult::from)
+      .thenAccept(r -> r.writeTo(routingContext.response()));
+  }
+  
+  private RequestAndRelatedRecords asMove(RequestAndRelatedRecords requestAndRelatedRecords,
+    JsonObject representation) {
+    String originalItemId = requestAndRelatedRecords.getItemId();    
+    String destinationItemId = representation.getString("destinationItemId");
+    if (representation.containsKey("requestType")) {
```

Comment:
I am not in favor of the approach of applying optional properties of a request body onto the request representation. This can be avoided by having the request body be of a request and if a property is desired to be changed it can during the move request. Additional properties, other than `requestType`, can be optionally changed if applicable. e.g. `fulfilmentPreference`, `pickupServicePoint`, etc. I can not say any of these are candidates for change during a move, but some might be and future properties of the schema may be as well.

The approach of having the request to be moved as the request body would require putting the `destinationItemId` in the path. These changes would be trivial and would like to open discussion on which approach should be taken.